src/controllers/task.list.controller.ts

```diff
@@ -1,28 +1,35 @@
-import { Request, Response } from "express";
+import { NextFunction, Request, Response } from "express";
 import { Templates } from "../types/template.paths";
 import { initTaskList, TaskList } from "../types/task.list";
 import { CompanyProfile } from "@companieshouse/api-sdk-node/dist/services/company-profile/types";
 import { getCompanyProfile } from "../services/company.profile.service";
 import { TRADING_STATUS_PATH, urlParams } from "../types/page.urls";
 import { isInFuture, toReadableFormat } from "../utils/date";
 
-export const get = async (req: Request, res: Response) => {
-  const companyNumber = req.params[urlParams.PARAM_COMPANY_NUMBER];
-  const backLinkUrl = TRADING_STATUS_PATH.replace(`:${urlParams.PARAM_COMPANY_NUMBER}`, companyNumber);
-  const company: CompanyProfile = await getCompanyProfile(companyNumber);
-  const taskList: TaskList = initTaskList();
-  taskList.recordDate = calculateFilingDate(taskList.recordDate, company);
-  return res.render(Templates.TASK_LIST, {
-    backLinkUrl,
-    company,
-    taskList
-  });
+export const get = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const companyNumber = req.params[urlParams.PARAM_COMPANY_NUMBER];
+    const backLinkUrl = TRADING_STATUS_PATH.replace(`:${urlParams.PARAM_COMPANY_NUMBER}`, companyNumber);
+    const company: CompanyProfile = await getCompanyProfile(companyNumber);
+    if (!company.confirmationStatement) {
+      throw new Error("No confirmation Statement data in company profile");
```

Comment:
i'd check with the analyst if we should show an error screen if the confirmation statement is missing. I think it might be missing from the company profile if it is the first year of the company and they haven't filed a CS before. In which case we might just want to show the current date? (analyst can advise)

Reply:
i just checked with an analyst and was told
> As soon as a company is incorporated, we generate the CS01 filing periods for them, and this info gets added to the Company_Profile delta then.
this means there will always be the dates needed, I will remove the check and error throw.
my confusion came from confirmationStatement being optional under the api-sdk-node, but from what ive been told that wouldnt ever be the case 🤷‍♂️

Reply:
The get function should still be wrapped by a try - catch (as you had it before but without the if (!company.confirmationStatement)  check). This will catch any errors thrown from the api calls and show the error page. If we don't have it then the page will just hang if any errors occur. All the controller get and post functions should be wrapped in a try catch, this one must have slipped through the net previously.